### DISCUSSION TITLE
PUT information on new device tab

### DISCUSSION THREAD

**Discussion by mobenchabane** [netbox-community/netbox#9966] [Q&A]
Hi All;

I'm new in the era of programin and coding
I'm coming from network world
 I am trying to write a code in Python that will use the API to get IP address of network devices (CISCO) and from there this programme will SSH all devices and get there config (Show running-config) until this point my code is working perfectly and I can print the confing on shell

My goal now is to  create a new tab on device and push the config that I got of each device in this new tab and I'm blocked at this point

If someone can help it will be great

Many thanks in advance

**Comment by zombah** [comment]
Hello,
I don't know about tab, but you can use your code to put config into html or git and then just create Custom Link button which points from device to it's config file.

**Reply by mobenchabane** [reply]
Hello Zombah;

many thanks for your reply.
what I read about Custom Link is that they point to another external tool like NMS, Monitoring ... but my goal is to store the config inside Netbox.

for exemple I want to put the device configuration that I got using my code in side "Config Context" tab of device but this didn't work for me

regards.

**Reply by zombah** [reply]
You can use Custom Script for commiting data into Netbox, so if i understand your idea try something like this - Run Custom Script-> Custom script connects to device get its config facts save them as vars-> Commit these facts into Netbox. This way you dont need to deal with extra tab.

**Reply by mobenchabane** [reply]
thanks zombah

I found that to creat a new tab we have to use plugin and this is the best way to get what I need, but after looking on how to creat a plugin it's not a simple think
so changed my strategy now I want just to push the config of devices that I got using my python script into "local_context_data" of each device the actual probleme that I have is the post/put request is not working for me and I belive that is because of the format that should be in json object format
